pkg/gatewayserver/gatewayserver_test.go

```diff
@@ -599,11 +606,19 @@ func TestGatewayServer(t *testing.T) {
 				case <-time.After(timeout):
 				}
 
-				ctx2, cancel2 := context.WithDeadline(ctx, time.Now().Add(timeout))
-				upCh := make(chan *ttnpb.GatewayUp)
-				downCh := make(chan *ttnpb.GatewayDown)
-				err := ptc.Link(ctx2, t, id, registeredGatewayKey, upCh, downCh)
-				cancel2()
+				var err error
+				for i := 0; i < 5; i++ {
+					ctx2, cancel2 := context.WithDeadline(ctx, time.Now().Add(timeout))
+					upCh := make(chan *ttnpb.GatewayUp)
+					downCh := make(chan *ttnpb.GatewayDown)
+					err = ptc.Link(ctx2, t, id, registeredGatewayKey, upCh, downCh)
+					cancel2()
+
+					if err.Error() != "error:unknown:unknown (context deadline exceeded)" {
+						break
+					}
+					time.Sleep(test.Delay)
+				}
```

Comment:
This seems like retrying the same thing.

Should the deadline timeout be longer?

Reply:
See comment above. Merely increasing the timeout does not solve this (I also got CI failures even with `2*timeout`)

Reply:
Then we have to increase it even more